domains/client/domain-executor/src/domain_block_processor.rs

```diff
@@ -406,7 +407,9 @@ where
 
         roots.push(state_root);
 
-        let trace_root = crate::merkle_tree::construct_trace_merkle_tree(roots.clone())?.root();
+        let trace_root = MerkleTree::from_leaves(&roots).root().ok_or_else(|| {
+            sp_blockchain::Error::Application(Box::from("Failed to get merkle root of trace"))
+        })?;
```

Comment:
Since we switch the library from `merkletree` to `rs_merkle` will there be any compatible issues we need to consider or do we need to wait for the next network reset?

Reply:
I think the network reset can be expected due to the following:

- There are already incompatible changes (#1148) merged into the main branch, and I believe there will be more incompatible changes coming from the New Messenger Model revamp.
-  W don't have to worry about the runtime API changes in the upcoming #1145.